src/main/scala/scorex/core/ModifiersCache.scala

```diff
@@ -0,0 +1,143 @@
+package scorex.core
+
+
+import scorex.core.consensus.HistoryReader
+import scorex.core.utils.ScorexLogging
+import scorex.core.validation.RecoverableModifierError
+
+import scala.collection.mutable
+import scala.util.{Failure, Success}
+
+/**
+  * A cache which is storing persistent modifiers not applied to history yet.
+  * @tparam PMOD - type of a persistent node view modifier (or a family of modifiers).
+  */
+trait ModifiersCache[PMOD <: PersistentNodeViewModifier, H <: HistoryReader[PMOD, _]] {
+  require(maxSize >= 1)
+
+  def size: Int = currentCacheSize
+
+  /**
+    * How many elements are to be stored in the cache
+    */
+  def maxSize: Int
+
+  type K = mutable.WrappedArray[Byte]
+  type V = PMOD
+
+  protected var currentCacheSize: Int = 0
```

Comment:
Why do you need this var?

Couldn't you simply use `cache.size` whenever you need to know the current cache size?

Reply:
done